jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -132,6 +132,14 @@ object XGBoost extends Serializable {
     try {
       val numEarlyStoppingRounds = params.get("num_early_stopping_rounds")
         .map(_.toString.toInt).getOrElse(0)
+      if (numEarlyStoppingRounds > 0) {
+        if (!params.contains("metrics_expected_direction") &&
+          (params("metrics_expected_direction") == "asc" ||
+            params("metrics_expected_direction") == "desc")) {
```

Comment:
Can you explain the logic here? I think we just need to check ```params.contains("metrics_expected_direction")```. If users don't set ```metrics_expected_direction``` explicitly, then ```params``` doesn't contain it.

Reply:
BTW, I saw there is param validation check at the definition:
```
final val metricsExpectedDirection = new Param[String](this, "metricsExpectedDirection",
    "the expected direction of the metrics' change, asc or desc",
    (value: String) => value == "asc" || value == "desc")
```
This will be called automatically with the two params setting method(MLlib way and XGBoost param map way), so inside the code, we don't need to validate params. I think here we just need to check ```params.contains("metrics_expected_direction")```. If contained, it must be legal.